vortex-array/src/arrays/list/compute/is_constant.rs

```
@@ -0,0 +1,134 @@
+use vortex_error::VortexResult;
+
+use crate::arrays::{ListArray, ListVTable};
+use crate::compute::{IsConstantKernel, IsConstantKernelAdapter, IsConstantOpts};
+use crate::register_kernel;
+
+impl IsConstantKernel for ListVTable {
+    fn is_constant(&self, array: &ListArray, opts: &IsConstantOpts) -> VortexResult<Option<bool>> {
+        // At this point, we're guaranteed:
+        // - Array has at least 2 elements
+        // - All elements are valid (no nulls)
+
+        if opts.is_negligible_cost() {
+            return Ok(None);
+        }
+
+        let first_list_len = array.offset_at(1) - array.offset_at(0);
```

Can you do something like `array.offsets().is_constant()`?

---

annoyingly offsets will not be constant for a constant list because of leading 0. You can slice off the first element

---

we need `offsets.diff().is_constant()` here right? We need some sort of `scan` reduction compute function to do this I think

---

actually I guess I can do something like
```
offsets_end = offsets.slice(1..)
binary_numeric(offset_end, offset, SUB).is_constant()
```
maybe that is better?

---

right, this will be incrementing by constant amount, so the delta has to be constant. we really want list view

---

I changed it to use compute functions now. We have lost the early exit but still might be better for some encodings

---

now we fail fast for the first 64 lists, and check the rest with compute functions